model/src/main/java/ru/training/karaf/repo/BookRepoImpl.java

```diff
@@ -0,0 +1,176 @@
+
+    public BookRepoImpl(JpaTemplate template) {
+        this.template = template;
+    }
+
+    @Override
+    public List<Book> getAll() {
+        return template.txExpr(em -> em.createNamedQuery(BookDO.GET_ALL, BookDO.class).getResultList())
+                .stream()
+                .map(BookImpl::new)
+                .collect(Collectors.toList());
+
+    }
+
+    @Override
+    public void create(String name, String usr, int price, int red, int green, int blue, int alpha, Collection<String> authors) {
+        BookDO bookToCreate = new BookDO();
+        ColorImpl color = new ColorImpl(red, green, blue, alpha);
+        bookToCreate.setName(name);
+        bookToCreate.setColor(color);
+
+
+        /*BookPriceRepoImpl bookPriceRepo = new BookPriceRepoImpl(template);
+        long l = bookPriceRepo.create(price);
+        bookToCreate.getPrice().add(template.txExpr(em -> em.find(BookPriceDO.class, l)));*/
+
+        bookToCreate.setAuthors(template.txExpr(em -> em.createNamedQuery(AuthorDo.GET_WHERE_IN, AuthorDo.class).setParameter("authors", authors)
+                .getResultList()));
+
+        UserDO userDO = template.txExpr(em -> em.createNamedQuery(UserDO.GET_BY_LOGIN, UserDO.class).setParameter("login", usr).getSingleResult());
+        bookToCreate.setUsr(userDO);
+
+        Long id = template.txExpr(em -> em.merge(bookToCreate)).getId();
+        BookPriceRepoImpl bookPriceRepo = new BookPriceRepoImpl(template);
+        bookPriceRepo.create(price, id);
+
+    }
+
+
+    @Override
+    public Optional<Book> get(long id) {
+        return template.txExpr(em -> getById(id, em)).map(BookImpl::new);
+    }
+
+    @Override
+    public void delete(String name) {
+        template.tx(em -> em.createNamedQuery(BookDO.DELETE_BY_NAME, BookDO.class).setParameter("name", name).executeUpdate());
+    }
+
+    @Override
+    public List<Book> getByUsr(String usr) {
+        return template.txExpr(em -> em.createNamedQuery(BookDO.GET_ALL_BY_USR, BookDO.class).setParameter("usr", usr).getResultList())
+                .stream()
+                .map(BookImpl::new)
+                .collect(Collectors.toList());
+    }
+
+    private Optional<BookDO> getById(long id, EntityManager em) {
+        try {
+            return Optional.of(em.createNamedQuery(BookDO.GET_BY_ID, BookDO.class).setParameter("id", id)
+                    .getSingleResult());
+        } catch (NoResultException e) {
+            return Optional.empty();
+        }
+    }
+
+    @Override
+    public void changePrice(String name, String author, int price) {
+        Optional<BookDO> bookDO = template.txExpr(em -> getByNameAndAuthor(em, name, author));
+        if (bookDO.isPresent()) {
+            BookDO book = bookDO.get();
+            BookPriceRepoImpl bookPriceRepo = new BookPriceRepoImpl(template);
+            bookPriceRepo.create(price, book.getId());
+        } else {
+            RequestLogRepoImpl requestLogRepo = new RequestLogRepoImpl(template);
```

Comment:
We should divide the responsibility between different layers and services. Repositories needed to interact with DB. We shouldn't burden them with business logic. So I think we should move this logic which decides if we update an existing book or make a log request to some other place.